src/settings/UIFeature.ts

```diff
@@ -0,0 +1,20 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// see settings.md for documentation on conventions
+export enum UIFeature {
+    URLPreviews = "UIFeature.URLPreviews",
```

Comment:
It feels a tad strange to me to use a setting key with a `.` in the name... Wouldn't that look strange to write in `config.json`? 

Does it mean you would write the following?

```
{
    "settingDefaults": {
        "UIFeature.URLPreviews": false
    }
}
```

The dot in the name suggests UI features form a logical group, so would it make more sense for them exist as object in the config?

```
{
    "settingDefaults": {
        "UIFeatures": {
            "URLPreviews": false
        }
    }
}
```

Aside from the structural question above, I think it would be more consistent with past settings for `UI` to be `ui` and `URL` to be `url` in the actual config file setting key.

Reply:
it does look a bit strange, however this is somewhat in line with how we (historically) group items. Structures are challenging and annoying to do in our settings system so we have done namespacing in several places. 

The namespacing mechanics (which seem to have disappeared from the documentation) are to UpperCamelCase the first part (`Widgets`, `RoomList`, `PinnedEvents`, etc) and lowerCamelCase the rest. The rough grouping is scoped to a feature, however in this case we are grouping features by features to provide a form of inline documentation in the config.

There's also an argument that we're more consistent with snake_case names, however this isn't consistent when using a namespace.

Overall I think the only change is to make it `url` instead of `URL` and leave the rest alone?

Reply:
Ah yes, I see there's also a pattern of `Widgets.foo` and such as you say. Yes, let's just change `URL` -> `url` and leave the rest as-is. Thanks for talking through it. 😄